java-frontend/src/main/java/org/sonar/java/model/DefaultJavaFileScannerContext.java

```diff
@@ -59,6 +59,10 @@
   private final JavaVersion javaVersion;
   private final boolean fileParsed;
 
+  private List<String> lines = null;
+  private List<String> linesWithEnding = null;
```

Comment:
Maybe we should consider protecting these lists from modification, as they will be used in many rules.